files/default/test-neutron-ha-tool.py

```diff
@@ -517,6 +522,97 @@ def test_setting_target_host_option(self):
         self.assertEqual('host', params.target_host)
 
 
+class TestSignalHandling(unittest.TestCase):
+
+    def wait_for_subprocess_to_write(self, subproc, text):
+        data = subproc.stdout.read(len(text))
+
+        self.assertEqual(text, data)
+
+    def test_term_signal_handling_functionality(self):
+        python = sys.executable
+        this_file = __file__
+        this_dir = os.path.dirname(this_file)
+
+        program = textwrap.dedent('''
+        import importlib
+        import sys
+        import time
+
+        ha_tool = importlib.import_module("neutron-ha-tool")
+        ha_tool.register_term_signal_handler()
+
+        with ha_tool.term_disabled():
+            sys.stdout.write('inside atomic section')
+            time.sleep(1)
+            sys.stdout.write('about to exit from atomic section')
+
+        time.sleep(10)
+        sys.exit(1)
+        ''')
```

Comment:
Actually I would use `multiprocessing` to create a different process to kill, not a temporal external python application